px4_ros2_cpp/src/navigation/experimental/local_navigation_interface.cpp

```diff
@@ -0,0 +1,201 @@
+  }
+
+  if (!_checkFrameValid(local_position_estimate)) {
+    return static_cast<int>(NavigationInterfaceCodes::ESTIMATE_FRAME_UNKNOWN);
+  }
+
+  if (local_position_estimate.timestamp_sample == 0) {
+    RCLCPP_WARN(_node.get_logger(), "Estimate sample timestamp is empty.");
+  }
+
+  // Populate aux local position message
+  AuxLocalPosition aux_local_position;
+
+  aux_local_position.timestamp_sample = local_position_estimate.timestamp_sample;
+
+  // Position
+  aux_local_position.pose_frame = _pose_frame;
+
+  const Vector2f position_xy = local_position_estimate.position_xy.value_or(Vector2f{NAN, NAN});
+  aux_local_position.position[0] = position_xy[0];
+  aux_local_position.position[1] = position_xy[1];
+  aux_local_position.position[2] = local_position_estimate.position_z.value_or(NAN);
+
+  const Vector2f position_xy_variance = local_position_estimate.position_xy_variance.value_or(
+    Vector2f{NAN, NAN});
+  aux_local_position.position_variance[0] = position_xy_variance[0];
+  aux_local_position.position_variance[1] = position_xy_variance[1];
+  aux_local_position.position_variance[2] =
+    local_position_estimate.position_z_variance.value_or(NAN);
+
+  // Attitude
+  const Quaternionf attitude_quaternion = local_position_estimate.attitude_quaternion.value_or(
+    Quaternionf{NAN, NAN, NAN, NAN});
+  aux_local_position.q[0] = attitude_quaternion.w();
+  aux_local_position.q[1] = attitude_quaternion.x();
+  aux_local_position.q[2] = attitude_quaternion.y();
+  aux_local_position.q[3] = attitude_quaternion.z();
+
+  const Vector3f attitude_variance = local_position_estimate.attitude_variance.value_or(
+    Vector3f{NAN, NAN, NAN});
+  aux_local_position.orientation_variance[0] = attitude_variance[0];
+  aux_local_position.orientation_variance[1] = attitude_variance[1];
+  aux_local_position.orientation_variance[2] = attitude_variance[2];
+
+  // Velocity
+  aux_local_position.velocity_frame = _velocity_frame;
+
+  const Vector2f velocity_xy = local_position_estimate.velocity_xy.value_or(Vector2f{NAN, NAN});
+  aux_local_position.velocity[0] = velocity_xy[0];
+  aux_local_position.velocity[1] = velocity_xy[1];
+  aux_local_position.velocity[2] = local_position_estimate.velocity_z.value_or(NAN);
+
+  const Vector2f velocity_xy_variance = local_position_estimate.velocity_xy_variance.value_or(
+    Vector2f{NAN, NAN});
+  aux_local_position.velocity_variance[0] = velocity_xy_variance[0];
+  aux_local_position.velocity_variance[1] = velocity_xy_variance[1];
+  aux_local_position.velocity_variance[2] =
+    local_position_estimate.velocity_z_variance.value_or(NAN);
+
+  // Angular velocity (unused at the moment)
+  aux_local_position.angular_velocity = {NAN, NAN, NAN};
+
+  // Publish
+  aux_local_position.timestamp = _node.get_clock()->now().nanoseconds() * 1e-3;
+  _aux_local_position_pub->publish(aux_local_position);
+
+  return static_cast<int>(NavigationInterfaceCodes::SUCCESS);
```

Comment:
@DanMesh just realized, maybe we should return a NavigationInterfaceCodes directly? It would prevent type casting, not sure if there's any advantage to returning an int?